File: android/ValleTPV/app/src/main/java/com/valleapp/valletpv/tools/CashlogyManager/InitializeAction.java
package com.valleapp.valletpv.tools.CashlogyManager;


public class InitializeAction extends CashlogyAction {

    public InitializeAction(CashlogySocketManager socketManager) {
        super(socketManager);
    }

    @Override
    public void execute() {
        // Enviar el comando #I# para inicializar la máquina
        socketManager.setCurrentAction(this);
        socketManager.sendCommand("#I#");
    }

    @Override
    public void handleResponse(String comando, String response) {

    }
}

File: android/ValleTPV/ValleTPVlib/src/main/java/com/valleapp/valletpvlib/cashlogymanager/InitializeAction.kt
package com.valleapp.valletpvlib.cashlogymanager

import android.os.Handler
import android.os.Looper
import android.util.Log

class InitializeAction(socketManager: CashlogySocketManager) :
    CashlogyAction(socketManager, Handler(Looper.getMainLooper())) {

    override fun execute() {
        socketManager.sendCommand("#I#", Handler(Looper.getMainLooper()) { msg ->
            Log.d("CASHLOGY", "Respuesta de #I#: ${msg.data.getString("value")}")
            return@Handler true
        })
    }
}
